### DISCUSSION TITLE
(HELP)FastAPI Testing : Sending bearer token as a Header

### DISCUSSION THREAD

**Discussion by naheedroomy** [fastapi/fastapi#4129] [Questions] [answered]
Hey guys, just implemented a protected API that relies on a token that I generate previously to make the endpoint work.

```
token = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoid"

def test_create_assignments():
    userassignments = {
        "email": "test@test.com",
        "skill": "string",
        "level": "string",
        "due_date": "string",
        "score": "string"
    }
    response = client.post("/user/assignments",
                           headers={"Authorization: Bearer ", token},
                           json=userassignments)
    assert response.status_code == 200
```

Can you tell me why this test keeps failing with the error "**_'set' object has no attribute 'items'_**"?

Thanks in advance!

**Comment by simondale00** [comment, marked as answer]
Hey @nroomy. Looks like you've got a typo with your `headers` kwarg. The provided value isn't a valid python dictionary; instead it's just written as two strings separated by a comma.

Changing `headers={"Authorization: Bearer", token}` to something like headers={"Authorization": f"Bearer {token}"}` should solve the problem for you.
